src/test/kotlin/edu/carroll/cs389/service/tracker/trackservice/TrackServiceQueryTest.kt

```diff
@@ -0,0 +1,561 @@
+    /* THESE TESTS ALL PERTAIN TO THE queryDatabase METHOD */
+    /*******************************************************/
+    /*******************************************************/
+
+    /*************************************/
+    /* Happy Path Tests (expected input) */
+    /*************************************/
+
+    @Test
+    fun validUserTest() {
+        val ipv4: String = "192.168.1.1"
+        val browser: String = "Firefox"
+        val browserMajorVersion: String = "110"
+        val platform: String = "Ubuntu"
+        val platformVersion: String = "22.04 LTS"
+        val requestURI: String = "/"
+
+        // logging time to be between
+        val timeU1i = Timestamp(Date().time)
+        Thread.sleep(100)
+        trackService.trackClient(ipv4, browser, browserMajorVersion, platform, platformVersion, requestURI)
+        Thread.sleep(100)
+        val timeU1f = Timestamp(Date().time)
+
+        val results = trackService.query()
+        assertTrue(
+            "validUserTest(): Tracked user should be equal",
+            results.size  == 1
+        )
+
+        assertTrue(
+            "validTrackedUserTest(): Tracked user should be equal to string information given " +
+                    "we set the timestamp of the inserted user to a known value",
+            ipv4 == results[0].clientIpv4Address() &&
+            "$browser $browserMajorVersion" == results[0].clientBrowserInfo() &&
+            "$platform $platformVersion" == results[0].clientOperatingSystem() &&
+            requestURI == results[0].clientConnectionRequestedPage() &&
+            timeU1i <= results[0].clientConnectionAttemptTimestamp() && 
+                    results[0].clientConnectionAttemptTimestamp() <= timeU1f
+        )
+    }
+
+    @Test
+    fun validTwoTrackedUserTest() {
+        val ipv4: String = "192.168.1.1"
+        val browser: String = "Firefox"
+        val browserMajorVersion: String = "110"
+        val platform: String = "Ubuntu"
+        val platformVersion: String = "22.04 LTS"
+        val requestURI: String = "/"
+
+        val ipv42: String = "172.98.272.47"
+        val browser2: String = "Chrome"
+        val browserMajorVersion2: String = "35"
+        val platform2: String = "Windows"
+        val platformVersion2: String = "10 NT"
+        val requestURI2: String = "/ip_info"
+
+        // logging time to be between
+        val timeU1i = Timestamp(Date().time)
+        Thread.sleep(100)
+        trackService.trackClient(ipv4, browser, browserMajorVersion, platform, platformVersion, requestURI)
+        Thread.sleep(100)
+        val timeU1f = Timestamp(Date().time)
+        
+        val timeU2i = Timestamp(Date().time)
+        Thread.sleep(100)
+        trackService.trackClient(ipv42, browser2, browserMajorVersion2, platform2, platformVersion2, requestURI2)
+        Thread.sleep(100)
+        val timeU2f = Timestamp(Date().time)
+
+        val results = trackService.query()
```

Comment:
One query, and then using the results.  Good (unlike the previous tests in the class above).